src/Symfony/Bridge/Doctrine/Form/ChoiceList/EntityChoiceList.php

```diff
@@ -108,15 +115,14 @@ public function __construct(ObjectManager $manager, $class, $labelPath = null, E
         $this->loaded = is_array($entities) || $entities instanceof \Traversable;
         $this->preferredEntities = $preferredEntities;
 
-        $identifier = $this->classMetadata->getIdentifierFieldNames();
+        $this->idField = $this->initIdentifierAs($this->classMetadata);
```

Comment:
I think the assignment and the following lines of code belong into the `initIdentifierAs()` method (which should be called `initIdentifierForClass()`). Methods should either change the state of the object or return a value. Doing both is dangerous.